toctoc/core/src/main/scala/io.buildo.toctoc/core/authentication/TokenBasedRecovery.scala

```diff
@@ -0,0 +1,54 @@
+package io.buildo.toctoc
+package core
+package authentication
+
+import cats.Monad
+import cats.data.EitherT
+
+import java.time.Duration
+
+object TokenBasedRecovery {
+  import TokenBasedAuthentication._
+
+  final case class TokenBasedRecoveryFlow[F[_]: Monad] private (
```

Comment:
Two doubts:
- why a case class?
- what's the purpose of the `object TokenBasedRecovery`? Namespacing?